packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -67,6 +67,63 @@ enum BottomNavigationBarType {
 ///    case it's assumed that each item will have a different background color
 ///    and that background color will contrast well with white.
 ///
+/// ## Sample Code
+///
+/// This example shows a [BottomNavigationBar] as it is used within a [Scaffold]
+/// widget. The [BottomNavigationBar] has three [BottomNavigationBarItem]
+/// widgets and the [currentIndex] is set to index 1. The color of the selected
+/// item is set to  a purple color. A function is called whenever any item is
+/// tapped and the function helps display the appropriate [Text] in the body of
+/// the [Scaffold].
+///
+/// ```dart
+/// class MyHomePage extends StatefulWidget {
+///  MyHomePage({Key key}) : super(key: key);
+///
+///  @override
+///  _MyHomePageState createState() => _MyHomePageState();
+/// }
+///
+/// class _MyHomePageState extends State<MyHomePage> {
+///  int _selectedIndex = 1;
+///  final _widgetOptions = [
+///    Text('Index 0: Home'),
+///    Text('Index 1: Business'),
+///    Text('Index 2: School'),
+///  ];
+///
+///  @override
+///  Widget build(BuildContext context) {
+///    return Scaffold(
+///      appBar: AppBar(
+///        title: Text('BottomNavigationBar Sample'),
+///      ),
+///      body: Center(
+///        child: _widgetOptions.elementAt(_selectedIndex),
+///      ),
+///      bottomNavigationBar: BottomNavigationBar(
+///        items: <BottomNavigationBarItem>[
+///          BottomNavigationBarItem(icon: Icon(Icons.home), title: Text('Home')),
+///          BottomNavigationBarItem(
+///              icon: Icon(Icons.business), title: Text('Business')),
+///          BottomNavigationBarItem(
+///              icon: Icon(Icons.school), title: Text('School')),
```

Comment:
nit: let's have all three BottomNavigationBarItem either be wrapped fully or all be on one line... it looks a bit weird with them differently wrapped.